docs/dsl/example.md

```diff
@@ -0,0 +1,438 @@
+
+Ultimately, we transform an `EventFrame`/`PatientFrame` into a `PatientSeries`, through filtering, sorting, aggregating, and selecting a column.
+
+A `PatientSeries` represents a collection of values with one value per patient.
+
+A `PatientSeries` can be assigned to a cohort, in which case a column with the `PatientSeries`'s values will be appended to the cohort's input.csv.
+
+---
+
+This diagram attempts to explain how a `EventFrame`/`PatientFrame` is transformed into a
+`PatientSeries`.
+
+The nodes are Python classes and the arrows are methods on those classes.
+
+```
+             +---+
+             |   V
+      filter |  PatientFrame
+             |   |   |
+             +---+   | select_column
+                     V
+                PatientSeries
+
+
+             +---+
+             |   V
+      filter |  EventFrame -----------------------+
+             |   |   |                            |
+             +---+   | sort_by                    |
+                     V                            |
+             SortedEventFrame                     |
+                     |                            |
+        +---+        | (first/last)_for_patient   | (count/exists)_for_patient
+        |   V        V                            |
+ filter |  AggregatedEventFrame                   |
+        |   |        |                            |
+        +---+        | select_column              |
+                     V                            |
+               PatientSeries <--------------------+
+```
+
+---
+
+Here's the code that we can use to generate an input.csv table.
+
+```
+from cohortextractor2.concepts.tables import (
+    immunisations as imms,
+    patients,
+    registrations,
+)
+from cohortextractor2.dsl import Cohort, categorise
+
+from codelists import covid_vacc_codes
+
+# Create a cohort.
+cohort = Cohort()
+
+# Create two PatientSeries for age and sex and assign to the cohort.
+cohort.age = patients.select_column(patients.age)
+cohort.sex = patients.select_column(patients.sex)
+
+# Create a new EventFrame representing just the events with covid vaccination codes.
+covid_imms = imms.filter(covid_vacc_codes.contains(imms.code))
+
+# Create an AggregatedEventFrame representing just, for each patient, the first record
+# in covid_imms.
+first = covid_imms.sort_by(imms.date).first_for_patient()
+
+# Create two PatientSeries for the first date / code and assign to the cohort.
+cohort.first_date = first.select_column(imms.date)
+cohort.first_code = first.select_column(imms.code)
+
+# Create an AggregatedEventFrame representing, for each patient, just the record that
+# occurs first 28 days after the first record of a covid vaccination.
+second = (
+    covid_imms.filter(imms.date > (cohort.first_date + 28))
+    .sort_by(imms.date)
+    .first_for_patient()
+)
+
+# Create two PatientSeries for the second date / code and assign to the cohort.
+cohort.second_date = second.select_column("date")
```

Comment:
question: What's the reason for using `"date"` as opposed to just `.date` in `imms.date` above?